tutorials/blitz/7_captum_explainability.py

```diff
@@ -0,0 +1,203 @@
+# provide explanations by extracting a subgraph and assigning importance
+# weights to the nodes and edges of it. We will visualize node-centered
+# weighted subgraphs through DGL's built-in functions. This is beneficial
+# for debugging and understanding GNNs and GNN explanation methods.
+
+# For this demonstration, we will use IntegratedGradients from `Captum <https://github.com/pytorch/captum>`__ to
+# explain the predictions of a graph convolutional network (GCN).
+# Specifically, we try to find the most important nodes and edges to the model in node classification.
+# Captum is a model interpretability and understanding library for PyTorch. You can install it with
+
+# .. code:: bash
+#
+# pip install captum
+#
+
+
+######################################################################
+# Loading Cora Dataset
+# --------------------
+#
+# First, we load DGL’s built-in Cora dataset and retrieve its graph structure, node labels (classes) and the number of node classes.
+# 
+
+# Install and import required packages.
+import dgl
+import dgl.data
+# The Cora dataset used in this tutorial only consists of one single graph.
+dataset = dgl.data.CoraGraphDataset()
+g = dataset[0]
+
+######################################################################
+# Define the model
+# ----------------
+# Then, we will build a two-layer Graph Convolutional Network (GCN).
+# Each layer computes new node representations by aggregating neighbor information.
+# What's more, we use GraphConv which supports ``edge_weight`` as a parameter to calculate
+# the importance of the edge in the edge explainability task.
+# 
+
+from dgl.nn import GraphConv
+# Define a class for GCN
+class GCN(nn.Module):
+    def __init__(self, in_feats, h_feats, num_classes):
+        super(GCN, self).__init__()
+        self.conv1 = GraphConv(in_feats, h_feats)
+        self.conv2 = GraphConv(h_feats, num_classes)
+
+    def forward(self, in_feat, g, edge_weight=None):
+        h = self.conv1(g, in_feat, edge_weight=edge_weight)
+        h = F.relu(h)
+        h = self.conv2(g, h, edge_weight=edge_weight)
+        return h
+
+######################################################################
+# Training the model
+# ------------------
+#
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+model = GCN(g.ndata['feat'].shape[1], 16, dataset.num_classes).to(device)
+g = g.to(device)
+optimizer = torch.optim.Adam(model.parameters(), lr=0.01, weight_decay=5e-4)
+
+features = g.ndata['feat']
+labels = g.ndata['label']
+train_mask = g.ndata['train_mask']
+
+for epoch in range(1,201):
+    model.train()
+    optimizer.zero_grad()
+    # Forward
+    logits = model(features, g)
+    
+    # Compute prediction
+    pred = logits.argmax(1)
+    
+    # Compute loss
+    # Note that you should only compute the losses of the nodes in the training set.
```

Comment:
This line may be not necessary as we can assume readers to already have some background on this topic.